python/ray/experimental/state/state_cli.py

```diff
@@ -35,6 +37,26 @@ class AvailableFormat(Enum):
     TABLE = "table"
 
 
+def _parse_filter(filter: str) -> Tuple[str, AvailablePredicate, SupportedFilterType]:
+    """Parse the filter string to a tuple of key, preciate, and value."""
+    # The function assumes there's going to be no key that includes "="" or "!=".
```

Comment:
should we throw exception if assumption violated?

Reply:
hmm I think it is a bit difficult. Right now, this assumes the value can contain `=` or `!=` (that's why we re-join the rest of the splits). I think it is only possible when we assume both key/value don't contain predicates values (which needs a bit more work)

Reply:
I will follow up on this in other PR (it will be much easier after we merge --detail PR)